### PR TITLE
python3Packages.dask: fix sandboxed builds

### PR DESCRIPTION
**Pull request by nbren12**
###### Motivation for this change

Allow `dask.dataframe` to be used in sandboxed builds. See #120307.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120310` at 031a8b7e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.clifford</li>
    <li>python38Packages.dask-jobqueue</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.rl-coach</li>
    <li>python39Packages.dask-jobqueue</li>
    <li>python39Packages.dask-xgboost</li>
    <li>python39Packages.rl-coach</li>
  </ul>
</details>
<details>
  <summary>16 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.caffe</li>
    <li>python38Packages.dask-glm</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.imgaug</li>
    <li>python38Packages.intake</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.stumpy</li>
    <li>python38Packages.sunpy</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>46 packages built successfully:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.batchgenerators</li>
    <li>python38Packages.dask</li>
    <li>python38Packages.dask-gateway</li>
    <li>python38Packages.dask-image</li>
    <li>python38Packages.dask-mpi</li>
    <li>python38Packages.distributed</li>
    <li>python38Packages.glymur</li>
    <li>python38Packages.image-match</li>
    <li>python38Packages.imagecorruptions</li>
    <li>python38Packages.pims</li>
    <li>python38Packages.pyfftw</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.radio_beam</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.scikitimage</li>
    <li>python38Packages.slicedimage</li>
    <li>python38Packages.sparse</li>
    <li>python38Packages.streamz</li>
    <li>python38Packages.stytra</li>
    <li>python38Packages.tensorly</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.batchgenerators</li>
    <li>python39Packages.dask</li>
    <li>python39Packages.dask-gateway</li>
    <li>python39Packages.dask-image</li>
    <li>python39Packages.dask-mpi</li>
    <li>python39Packages.distributed</li>
    <li>python39Packages.glymur</li>
    <li>python39Packages.image-match</li>
    <li>python39Packages.imagecorruptions</li>
    <li>python39Packages.intake</li>
    <li>python39Packages.pims</li>
    <li>python39Packages.pyfftw</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.radio_beam</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.scikitimage</li>
    <li>python39Packages.slicedimage</li>
    <li>python39Packages.streamz</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/python-modules/dask/default.nix:55:15:
```
   |
55 |   patches = [ ./0001-guard-blocksize-inference-logic.patch ];
   |               ^
```
</li>
  </ul>
</details>

**Comment by FRidh** [comment]
Did you report the issue upstream? If not, please do so, submit the patch, and fetch it instead of including directly in Nixpkgs.

**Comment by nbren12** [comment]
I opened a PR upstream: https://github.com/dask/dask/pull/7601.

Will I need to wait for this to be merged before using it here?

**Comment by FRidh** [comment]
> Will I need to wait for this to be merged before using it here?

No need for that. Please use `fetchpatch` instead of keeping the patch in the tree.

**Comment by nbren12** [comment]
>Please use fetchpatch 

I just did this.

One question: can github possibly garbage collect a patch if the upstream PR uses a merge strategy that modifies the history (e.g. rebase or squash)?

**Comment by SuperSandro2000** [comment]
> One question: can github possibly garbage collect a patch if the upstream PR uses a merge strategy that modifies the history (e.g. rebase or squash)?

It shouldn't https://github.community/t/does-github-ever-purge-commits-or-files-that-were-visible-at-some-time/1944/2?u=supersandro2000.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 120310` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.clifford</li>
<li>python38Packages.dask-jobqueue</li>
<li>python38Packages.dask-xgboost</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.dask-jobqueue</li>
<li>python39Packages.dask-xgboost</li>
<li>python39Packages.rl-coach</li>
</ul>
</details>
<details>

  <summary>4 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python38Packages.imgaug: log was empty</li>
<li>python38Packages.mask-rcnn: log was empty</li>
<li>python39Packages.imgaug: log was empty</li>
<li>python39Packages.mask-rcnn: log was empty</li>
  </ul>
</details>
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>python39Packages.streamz: log was empty</li>
  </ul>
</details>
<details>
<summary>
57 packages built:
</summary>
<ul>
<li>python38Packages.aplpy</li>
<li>python38Packages.asdf</li>
<li>python38Packages.batchgenerators</li>
<li>python38Packages.caffe</li>
<li>python38Packages.dask</li>
<li>python38Packages.dask-gateway</li>
<li>python38Packages.dask-glm</li>
<li>python38Packages.dask-image</li>
<li>python38Packages.dask-ml</li>
<li>python38Packages.dask-mpi</li>
<li>python38Packages.datashader</li>
<li>python38Packages.distributed</li>
<li>python38Packages.glymur</li>
<li>python38Packages.image-match</li>
<li>python38Packages.imagecorruptions</li>
<li>python38Packages.intake</li>
<li>python38Packages.pims</li>
<li>python38Packages.pyfftw</li>
<li>python38Packages.pyregion</li>
<li>python38Packages.pytest-astropy</li>
<li>python38Packages.pytest-astropy-header</li>
<li>python38Packages.radio_beam</li>
<li>python38Packages.reproject</li>
<li>python38Packages.scikitimage</li>
<li>python38Packages.slicedimage</li>
<li>python38Packages.sparse</li>
<li>python38Packages.spectral-cube</li>
<li>python38Packages.streamz</li>
<li>python38Packages.stumpy</li>
<li>python38Packages.stytra</li>
<li>python38Packages.sunpy</li>
<li>python38Packages.tensorly</li>
<li>python39Packages.aplpy</li>
<li>python39Packages.asdf</li>
<li>python39Packages.batchgenerators</li>
<li>python39Packages.caffe</li>
<li>python39Packages.dask</li>
<li>python39Packages.dask-gateway</li>
<li>python39Packages.dask-glm</li>
<li>python39Packages.dask-image</li>
<li>python39Packages.dask-mpi</li>
<li>python39Packages.distributed</li>
<li>python39Packages.glymur</li>
<li>python39Packages.image-match</li>
<li>python39Packages.imagecorruptions</li>
<li>python39Packages.intake</li>
<li>python39Packages.pims</li>
<li>python39Packages.pyfftw</li>
<li>python39Packages.pyregion</li>
<li>python39Packages.pytest-astropy</li>
<li>python39Packages.pytest-astropy-header</li>
<li>python39Packages.radio_beam</li>
<li>python39Packages.reproject</li>
<li>python39Packages.scikitimage</li>
<li>python39Packages.slicedimage</li>
<li>python39Packages.spectral-cube</li>
<li>python39Packages.sunpy</li>
</ul>
</details>
